components/speedreader/speedreader_throttle.cc

```diff
@@ -61,6 +101,12 @@ void SpeedReaderThrottle::WillProcessResponse(
     // Skip all non-html documents.
     return;
   }
+
+  if (response_head) {
+    SetSpeedreaderCSP(response_url, response_head,
```

Comment:
we need to update headers only if the content has been rewritten, as in mime sniffer